source/jormungandr/jormungandr/scenarios/helper_classes/streetnetwork_path.py

```diff
@@ -111,8 +202,10 @@ def _do_request(self):
         )
 
         dp = self._direct_path_with_fp(self._streetnetwork_service)
-        prepend_first_coord(dp, self._orig_obj)
-        append_last_coord(dp, self._dest_obj)
+        origin = self.get_origin_or_destination(dp, StreetNetworkPathType.BEGINNING_FALLBACK)
+        destination = self.get_origin_or_destination(dp, StreetNetworkPathType.ENDING_FALLBACK)
```

Comment:
hhhhhhhhhm..... this smells fishy... A direct_path can be either `BEGINNING_FALLBACK`, `ENDING_FALLBACK` or `DIRECT`... 

you should be able to determine `origin` and `destination` based on `self._path_type`... 

let's talk if it's not clear...

Reply:
Done